### DISCUSSION TITLE
Performance Degradation in proxySet Instances with Increased Keys.

### DISCUSSION THREAD

**Discussion by kk8787** [pmndrs/valtio#853] [Bug report]
## Summary

Performance Degradation in proxySet Instances with Increased Keys.
I checked the source code, and the performance deteriorates linearly because it performs a linear search.

## Link to reproduction

When using methods such as get and set.
In my environment, it becomes quite slow with over 5000 keys.

## Check List

Please do not ask questions in issues.

- [ ✓] I've already opened a [discussion](https://github.com/pmndrs/valtio/discussions) before opening this issue, or already discussed in other media.

Please include a minimal reproduction.

- [ ✓] I've added a link to a [typescript playground](https://www.typescriptlang.org/play) or [codesandbox](https://codesandbox.io) with a minimal reproduction.

**Comment by dai-shi** [comment]
That's expected, because proxySet is just an emulation.
Or, do you have any idea to improve the performance without breaking the capability? (Maybe, a tree like structure would do, but the implementation would be challenging.)

**Reply by kk8787** [reply]
I see, it's not a simple problem.
My use case allows me to use plain objects instead.
thank you
